community/cypher/acceptance/src/test/scala/org/neo4j/internal/cypher/acceptance/ShortestPathComplexQueryAcceptanceTest.scala

```diff
@@ -0,0 +1,92 @@
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.internal.cypher.acceptance
+
+import org.hamcrest.CoreMatchers._
+import org.junit.Assert._
+import org.neo4j.cypher.internal.RewindableExecutionResult
+import org.neo4j.cypher.internal.compatibility.ExecutionResultWrapperFor3_0
+import org.neo4j.cypher.{ExecutionEngineFunSuite, NewPlannerTestSupport}
+import org.neo4j.graphdb.Node
+
+class ShortestPathComplexQueryAcceptanceTest extends ExecutionEngineFunSuite with NewPlannerTestSupport {
+
+  test("allShortestPaths with complex LHS should be planned with exhaustive fallback and include predicate") {
+    setupModel()
+    val result = executeUsingCostPlannerOnly(
+      """
+        |//same idea but iterating over two collections of nodes
+        |profile match (charles:Pixie { fname : 'Charles'}),(joey:Pixie { fname : 'Joey'}),(kim:Pixie { fname : 'Kim'})
+        |with kim as kimDeal, collect(charles) as charlesT, collect(joey) as joeyS
+        |unwind charlesT AS charlesThompson
+        |unwind joeyS AS joeySantiago
+        |match pathx = allShortestPaths((charlesThompson)-[*1..5]-(joeySantiago))
+        |where none (n IN nodes(pathx) where id(n) = id(kimDeal))
+        |with nodes(pathx) as nodes
+        |return nodes
+      """.stripMargin)
+    val results = result.columnAs("nodes").toList
+    println(results)
+    println(result.executionPlanDescription())
+    val ids = results(0).asInstanceOf[Seq[_]].map(n => n.asInstanceOf[Node].getId)
+    ids should be(List(0, 4, 3, 2))
+    result should use("VarLengthExpand(Into)", "AntiConditionalApply")
+  }
+
+  test("shortestPath with complex LHS should be planned with exhaustive fallback and include predicate") {
+    System.setProperty("pickBestPlan.VERBOSE","true")
+    setupModel()
+    val result = executeUsingCostPlannerOnly(
+      """
+        |profile match (charles:Pixie { fname : 'Charles'}),(joey:Pixie { fname : 'Joey'}),(kim:Pixie { fname : 'Kim'})
+        |with kim as kimDeal, collect(charles) as charlesT, collect(joey) as joeyS
+        |unwind charlesT AS charlesThompson
+        |unwind joeyS AS joeySantiago
+        |match pathx = shortestPath((charlesThompson)-[*1..5]-(joeySantiago))
+        |where none (n IN nodes(pathx) where id(n) = id(kimDeal))
+        |unwind nodes(pathx) as nodes
+        |return nodes
+      """.stripMargin)
+    val results = result.columnAs("nodes").toList
+    println(results)
+    println(result.executionPlanDescription())
+    assertThat(results.length, equalTo(4))
```

Comment:
That's a pretty vague result assertion; why not assert on the exact result contents?